File: frontend/src/game/objects/fire.ts
import { COUNTDOWN_MS, TILE_WIDTH } from "../gameConfig";
import { ANIMS } from "../sprite_animations/animations";
import { ObjType } from "../types";
export interface FireObjConfig {
    centerX: number,
    centerY: number,
    verticalStart: number,
    verticalEnd: number,
    horizontalStart: number,
    horizontalEnd: number

}
export class FireObj {
    type: ObjType = "fire"
    verticalSprites: Phaser.GameObjects.Sprite[]
    horizontalSprites: Phaser.GameObjects.Sprite[]
    centerSprite: Phaser.GameObjects.Sprite
    remainingMs: number = COUNTDOWN_MS.fire

    constructor(
        scene: Phaser.Scene,
        config: FireObjConfig
    ) {
        const { centerX, centerY, verticalEnd, verticalStart, horizontalEnd, horizontalStart } = config
        this.centerSprite = scene.add.sprite(centerX * TILE_WIDTH, centerY * TILE_WIDTH, 'explosion')
        this.centerSprite.setOrigin(0, 0)
        this.centerSprite.setDepth(2)
        this.centerSprite.anims.play(ANIMS.explosion.start)

        this.verticalSprites = []
        this.horizontalSprites = []

        // Up
        for (let i = 1; i <= verticalStart; i++) {
            const sprite = scene.add.sprite(centerX * TILE_WIDTH, (centerY - i) * TILE_WIDTH, 'explosion')
            sprite.setOrigin(0, 0)
            sprite.setDepth(2)

            sprite.anims.play(i === verticalStart ? ANIMS.explosion.up.end : ANIMS.explosion.up.middle)
            this.verticalSprites.push(sprite)
        }

        // Down
        for (let i = 1; i <= verticalEnd; i++) {
            const sprite = scene.add.sprite(centerX * TILE_WIDTH, (centerY + i) * TILE_WIDTH, 'explosion')
            sprite.setOrigin(0, 0)
            sprite.setDepth(2)
            sprite.anims.play(i === verticalEnd ? ANIMS.explosion.down.end : ANIMS.explosion.down.middle)
            this.verticalSprites.push(sprite)
        }

        // Left
        for (let i = 1; i <= horizontalStart; i++) {
            const sprite = scene.add.sprite((centerX - i) * TILE_WIDTH, centerY * TILE_WIDTH, 'explosion')
            sprite.setOrigin(0, 0)
            sprite.setDepth(2)
            sprite.anims.play(i === horizontalStart ? ANIMS.explosion.left.end : ANIMS.explosion.left.middle)
            this.horizontalSprites.push(sprite)
        }

        // Right
        for (let i = 1; i <= horizontalEnd; i++) {
            const sprite = scene.add.sprite((centerX + i) * TILE_WIDTH, centerY * TILE_WIDTH, 'explosion')
            sprite.setOrigin(0, 0)
            sprite.setDepth(2)
            sprite.anims.play(i === horizontalEnd ? ANIMS.explosion.right.end : ANIMS.explosion.right.middle)
            this.horizontalSprites.push(sprite)
        }
    }

    destroy() {
        this.centerSprite.destroy()
        this.verticalSprites.forEach(s => s.destroy())
        this.horizontalSprites.forEach(s => s.destroy())
    }
}

File: py-game/game/objects/fire.py
from config import COUNTDOWN_FIRE_MS, TILE_WIDTH


class FireObj:
    """
    A fire explosion instance.
    Stores tile indices for the center and each arm segment.
    """
    def __init__(self, center_x: int, center_y: int,
                 vertical_start: int, vertical_end: int,
                 horizontal_start: int, horizontal_end: int):
        self.obj_type = "fire"
        self.remaining_ms = COUNTDOWN_FIRE_MS

        # All tile positions this fire occupies: list of (ix, iy)
        self.tiles: list[tuple[int, int]] = []
        self.tiles.append((center_x, center_y))

        for i in range(1, vertical_start + 1):
            self.tiles.append((center_x, center_y - i))
        for i in range(1, vertical_end + 1):
            self.tiles.append((center_x, center_y + i))
        for i in range(1, horizontal_start + 1):
            self.tiles.append((center_x - i, center_y))
        for i in range(1, horizontal_end + 1):
            self.tiles.append((center_x + i, center_y))
